apps/studio/src/features/editing-experience/components/RootStateDrawer.tsx

```diff
@@ -24,20 +27,58 @@ export default function RootStateDrawer() {
     setSavedPageState,
     setPreviewPageState,
   } = useEditorDrawerContext()
+  const router = useRouter()
+  const pageId = Number(router.query.pageId)
+  const siteId = Number(router.query.siteId)
```

Comment:
Same as above, suggest to take this in as a prop also instead of retrieving from query params, since we don't want to implicitly tie this component to the page.
can see other examples of `useQueryParse` in the current codebase.

```ts
// in some page/path/to/page.ts
const {pageId, siteId} = useQueryParse(editPageSchema)

// Pass into this RootStateDrawer (or maybe even EditorDrawerContext)
```

Reply:
updated - using the one you've done in the base page. i felt that made more sense than defining my own because this is a component on the page itself